src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,852 @@
+ * variant versus a sequencing or data processing artifact. This model is determined adaptively based on "true sites" provided
+ * as input, typically HapMap 3 sites and those sites found to be polymorphic on the Omni 2.5M SNP chip array (in humans). This
+ * adaptive error model can then be applied to both known and novel variation discovered in the call set of interest to evaluate
+ * the probability that each call is real. The score that gets added to the INFO field of each variant is called the VQSLOD. It is
+ * the log odds of being a true variant versus being false under the trained Gaussian mixture model.
+ * </p>
+ *
+ * <p>
+ * This tool performs the first pass in a two-stage process called VQSR; the second pass is performed by the
+ * <a href='https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.php'>ApplyRecalibration</a> tool.
+ * In brief, the first pass consists of creating a Gaussian mixture model by looking at the distribution of annotation
+ * In brief, the first pass consists of creating a Gaussian mixture model by looking at the distribution of annotation
+ * values over a high quality subset of the input call set, and then scoring all input variants according to the model.
+ * The second pass consists of filtering variants based on score cutoffs identified in the first pass.
+ *</p>
+ *
+ * <p>VQSR is probably the hardest part of the Best Practices to get right, so be sure to read the
+ * <a href='https://www.broadinstitute.org/gatk/guide/article?id=39'>method documentation</a>,
+ * <a href='https://www.broadinstitute.org/gatk/guide/article?id=1259'>parameter recommendations</a> and
+ * <a href='https://www.broadinstitute.org/gatk/guide/article?id=2805'>tutorial</a> to really understand what these
+ * tools and how to use them for best results on your own data.</p>
+ *
+ * <h3>Inputs</h3>
+ * <ul>
+ * <li>The input raw variants to be recalibrated. These variant calls must be annotated with the annotations that will be
+ * used for modeling. If the calls come from multiple samples, they must have been obtained by joint calling the samples,
+ * either directly (running HaplotypeCaller on all samples together) or via the GVCF workflow (HaplotypeCaller with -ERC
+ * GVCF per-sample then GenotypeGVCFs on the resulting gVCFs) which is more scalable.</li>
+ * <li>Known, truth, and training sets to be used by the algorithm. See the method documentation for more details.</li>
+ * </ul>
+ *
+ * <h3>Output</h3>
+ * <ul>
+ * <li>A recalibration table file that will be used by the ApplyRecalibration tool.</li>
+ * <li>A tranches file which shows various metrics of the recalibration callset for slices of the data.</li>
+ * </ul>
+ *
+ * <h3>Usage example</h3>
+ * <p>Recalibrating SNPs in exome data:</p>
+ * <pre>
+ * ./gatk-launch JAVA_OPTS=-Xmx4g \
```

Comment:
A definite TODO is to confirm that our version actually does run in 4g with real inputs...